Cloud changelog - April 21 (#4482)
mattermost/docs: source/cloud/cloud-administration/cloud-changelog.md

This changelog summarizes updates to [Mattermost Cloud](https://mattermost.com/get-started/), an enterprise-grade SaaS offering hosted by Mattermost.

## Release 2021-04-21

### Highlights

#### Apps Framework (Developer Preview)

>  - This feature is available on Mobile Apps when Apps Framework becomes available on Mattermost v5.35.0 and the proxy plugin is loaded on an instance. The launch for the Developer Preview of the Apps Framework is scheduled for April 29th.

Reviewer: ```suggestion
 - Apps Framework is a new way to integrate with external tools, and allows developers to create interactive Apps in Mattermost, using any development language they're comfortable with. The new Apps work seamlessly across mobile and desktop clients.  This is a developer preview and is not intended for production instances of Mattermost yet.  This feature will be available for self-managed customers when Mattermost v5.35.0 ships and the Apps Framework Plugin is loaded on an instance, before then developers can use the `cloud` branch to get a local test environment running. The launch for the Developer Preview of the Apps Framework is scheduled for April 29th.  Learn more: 
```

Reviewer: @justinegeffen - do we have the URL for the new Apps Framework docs location?  (want to insert it here)

Reviewer 2: It should be located at https://developers.mattermost.com/apps/getting-started/ once it's live but if we could hold off on that until I've ironed out some issues that'd be great.

This changelog summarizes updates to [Mattermost Cloud](https://mattermost.com/get-started/), an enterprise-grade SaaS offering hosted by Mattermost.

## Release 2021-04-21

### Highlights

#### Apps Framework, lightweight interactive add-ons to Mattermost

>  - 

Reviewer 3: Should we add something similar to what we did in mobile?
```
This feature is available on Mobile Apps when Apps Framework becomes available on Mattermost v5.35.0 and the proxy plugin is loaded on an instance. The launch for the Developer Preview of the Apps Framework is scheduled for April 29th.
```
Probably will need some rephrasing for this context. @aaronrothschild what do you think?

Author: I added a description.

#### Files Search

>  - Users can now search file attachments by name and document contents. This feature is available on Mobile Apps in a future release.

Reviewer 4: Searching in Mattermost now finds both relevant messages and files in your team's conversation history. Search will return results for attachments that match the file name or contain matching text content within supported document types. [Learn more](https://mattermost.com/blog/file-search/)

#### Migrated Experimental, About, Reporting, Environment, Site Configuration, Authentication, Integrations, Compliance sections to their respective sub-section permissions.
 - 

#### Shared Channels: Experimental support added for sharing channels between Mattermost clusters. Requires an E20 license. The Shared Channels feature is disabled by default
 - 

#### Enterprise Trial enhancements Phase 1
 - 

### Improvements

#### User Interface (UI)
 - Added support for allowing collapsing in-line images over 100px in height.
 - Implemented maximum length validation on the status modal for custom statuses.
 - Synchronized collapsed channel sidebar categories on the server.
 - Empty state is no longer off-centered in the **Channel Switcher**.
 - Ephemeral message created from call response ``markdown`` field is now posted by bot.

#### Administration
 - Added a command line document extraction command that allows indexing documents by content.

### Bug Fixes
 - Fixed link previews on a number of websites, including Reddit.

>  - Fixed an issue where SAML assigned Mattermost UserID as username if the value was invalid and did not log this.

Reviewer 5: ```suggestion
 - Fixed an issue where SAML assigned Mattermost ``UserID`` as username if the value was invalid and did not log this.
```

 - Fixed an issue where hover effects for category sorting and **Direct Messages** category limit submenus were too dark on a dark theme.

>  - Fixed an issue where users were unable to drag vertical scroll bar on a PDF preview.

Reviewer 5: ```suggestion
 - Fixed an issue where users were unable to drag the vertical scroll bar on a PDF preview.
```

This changelog summarizes updates to [Mattermost Cloud](https://mattermost.com/get-started/), an enterprise-grade SaaS offering hosted by Mattermost.

## Release 2021-04-21

### Highlights

#### Apps Framework (Developer Preview)
 - This feature is available on Mobile Apps when Apps Framework becomes available on Mattermost v5.35.0 and the proxy plugin is loaded on an instance. The launch for the Developer Preview of the Apps Framework is scheduled for April 29th.

#### Files Search
 - Searching in Mattermost now finds both relevant messages and files in your team's conversation history. Search will return results for attachments that match the file name or contain matching text content within supported document types. [Learn more](https://mattermost.com/blog/file-search/).

#### Granular Access to System Console Pages

>  - Migrated Experimental, About, Reporting, Environment, Site Configuration, Authentication, Integrations, Compliance sections to their respective sub-section permissions.

Reviewer 6: @sbishel 

We should mention here that upgrading to this version will cause problems if the user decides to downgrade to an older version of Mattermost due to the permission deletion. 
As we discussed in our meeting, the best course of action is coming up with script that does the rollback.

Anyways just mentioning you on this to give visibility if there is anymore you'd like to add.

Author: I'll add a note to v5.35 changelog, likely not needed for the Cloud changelog.

Reviewer 7: We reverted that change, so it is not needed in v5.35 change log either.

#### Shared Channels (Experimental)
 - Experimental support was added for sharing channels between Mattermost clusters. This feature is disabled by default.

#### Enterprise Trial Enhancements Phase 1

>  - 

Reviewer 8: Please remove this section about Trial enhancements. It doesn't pertain to Cloud instances.

This changelog summarizes updates to [Mattermost Cloud](https://mattermost.com/get-started/), an enterprise-grade SaaS offering hosted by Mattermost.

## Release 2021-04-21

### Highlights

#### Apps Framework (Developer Preview)

>  - Apps Framework is a new way to integrate with external tools, and allows developers to create interactive Apps in Mattermost, using any development language they're comfortable with. The new Apps work seamlessly across mobile and desktop clients. This is a developer preview and is not intended for production instances of Mattermost yet. This feature will be available for self-managed customers when Mattermost v5.35.0 ships and the Apps Framework Plugin is loaded on an instance, before then developers can use the `cloud` branch to get a local test environment running. The launch for the Developer Preview of the Apps Framework is scheduled for April 29th. Learn more: https://developers.mattermost.com/integrate/apps/.

Reviewer 5: ```suggestion
 - The Mattermost Apps Framework introduces a new way to integrate with external tools to allow developers to create interactive Apps in Mattermost using any development language they're comfortable with. These Apps work seamlessly across mobile and desktop clients. This is a developer preview that is not yet intended for production instances of Mattermost. The Apps Framework will be available for self-managed customers in Mattermost v5.35 once the Apps Framework Plugin is loaded on an instance. Until then, developers can use the `cloud` branch to get a local test environment running. The launch for the Developer Preview of the Apps Framework is scheduled for April 29th, 2021. Learn more: https://developers.mattermost.com/integrate/apps/.
```

>  - Apps Framework is a new way to integrate with external tools, and allows developers to create interactive Apps in Mattermost, using any development language they're comfortable with. The new Apps work seamlessly across mobile and desktop clients. This is a developer preview and is not intended for production instances of Mattermost yet. This feature will be available for self-managed customers when Mattermost v5.35.0 ships and the Apps Framework Plugin is loaded on an instance, before then developers can use the `cloud` branch to get a local test environment running. The launch for the Developer Preview of the Apps Framework is scheduled for April 29th. Learn more: https://developers.mattermost.com/apps/getting-started/.

Reviewer 2: ```suggestion
 - Apps Framework is a new way to integrate with external tools, and allows developers to create interactive Apps in Mattermost, using any development language they're comfortable with. The new Apps work seamlessly across mobile and desktop clients. This is a developer preview and is not intended for production instances of Mattermost yet. This feature will be available for self-managed customers when Mattermost v5.35.0 ships and the Apps Framework Plugin is loaded on an instance, before then developers can use the `cloud` branch to get a local test environment running. The launch for the Developer Preview of the Apps Framework is scheduled for April 29th. Learn more: https://developers.mattermost.com/integrate/apps/.
```

#### Files Search
 - Searching in Mattermost now finds both relevant messages and files in your team's conversation history. Search will return results for attachments that match the file name or contain matching text content within supported document types. [Learn more](https://mattermost.com/blog/file-search/).

#### Granular Access to System Console Pages

>  - Migrated Experimental, About, Reporting, Environment, Site Configuration, Authentication, Integrations, Compliance sections to their respective sub-section permissions.

Reviewer 5: ```suggestion
 - Migrated the following System Console sections to their respective sub-section permissions: Experimental, About, Reporting, Environment, Site Configuration, Authentication, Integrations, and Compliance.
```

#### Shared Channels (Experimental)
 - Experimental support was added for sharing channels between Mattermost clusters. This feature is disabled by default.

### Improvements

#### User Interface (UI)
 - Added support to collapse in-line images over 100px in height.
 - Implemented maximum length validation on the status modal for custom statuses.
 - Synchronized collapsed channel sidebar categories on the server.
 - Empty state is no longer off-centered in the **Channel Switcher**.
 - Ephemeral message created from call response ``markdown`` field is now posted by bot.

>  - Added improvements and fixes for the custom status feature, such as fixed an issue where recently selected statuses were missing from the **Set a Status** confirmation screen and updated **Mobile Push Notifications** text in **Account Settings** to refer to user **availability** instead of **online status**.

Reviewer 5: ```suggestion
 - Added improvements and fixes for the custom status feature. Fixed an issue where recently selected statuses were missing from the **Set a Status** confirmation screen, and updated the **Mobile Push Notifications** text in **Account Settings** to refer to user **availability** instead of **online status**.
```

>  - Moved the user status in the channel switcher to overlap with avatar and added a URL 'Slug' information to channel names in Channel Switcher.

Reviewer 5: ```suggestion
 - Moved the user status in the channel switcher to overlap with user avatars, and added URL 'Slug' information to channel names in the channel switcher.
```

#### Administration
 - Paused admin advisor notifications from triggering.
 - Added a command line document extraction command that allows indexing documents by content.

### Bug Fixes
 - Fixed link previews on a number of websites, including Reddit.
 - Fixed an issue where SAML assigned Mattermost ``UserID`` as username if the value was invalid and did not log this.
 - Fixed an issue where hover effects for category sorting and **Direct Messages** category limit submenus were too dark on a dark theme.
 - Fixed an issue where users were unable to drag the vertical scroll bar on a PDF preview.
 - Fixed an issue with animations on long posts when highlighted as a permalink.

>  - Fixed an issue where the user nickname was not shown on channel switch.

Reviewer 5: ```suggestion
 - Fixed an issue where the user's nickname was not shown on channel switch.
```

 - Fixed an issue where deactivated users were not marked as "Deactivated" in the channel switcher.

>  - Fixed an issue where queries executed during the upgrade process would preemptively timeout on the application side.

Reviewer 5: ```suggestion
 - Fixed an issue where queries executed during the upgrade process would preemptively time out on the application side.
```
